opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/core/ExpressionScript.java

```diff
@@ -107,7 +113,12 @@ public Object visitParse(ParseExpression node, Set<ReferenceExpression> context)
 
   private OpenSearchExprValueFactory buildValueFactory(Set<ReferenceExpression> fields) {
     Map<String, OpenSearchDataType> typeEnv = fields.stream().collect(toMap(
-        ReferenceExpression::getAttr, e -> OpenSearchDataType.of(e.type())));
+        ReferenceExpression::getAttr, e -> {
+          if (OpenSearchDateType.isDateTypeCompatible(e.type())) {
+            return OpenSearchDateType.of(e.type());
```

Comment:
The same - move compatibility check into `of` method

Reply:
Done in https://github.com/Bit-Quill/opensearch-project-sql/pull/268/commits/58a8f193ffe1407665734d55a60e53ff9bc97aa8